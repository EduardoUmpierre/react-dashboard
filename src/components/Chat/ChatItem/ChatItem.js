import React from 'react'
import './ChatItem.scss'

const chatItem = ({
    displayPortraitLeft,
    userName,
    portrait,
    time,
    message
}) => {
    let classes = ['chat-item']

    if (displayPortraitLeft) {
        classes.push('chat-item--left')
    }

    return (
        <div className={classes.join(' ')}>
            <div className='chat-item__photo'>
                {portrait && <img src={portrait} alt={userName} />}
            </div>

            <div className='chat-item__data'>
                <div className='chat-item__upper'>
                    <div className='chat-item__name'>{userName}</div>
                    <div className='chat-item__time'>{time}</div>
                </div>
                <p className='chat-item__message'>{message}</p>
            </div>
        </div>
    )
}

export default chatItem
