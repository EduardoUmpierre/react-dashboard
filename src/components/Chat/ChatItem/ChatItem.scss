.chat-item {
    display: flex;
    padding: 1.563rem 0 0.938rem;

    & + & {
        border-top: 1px solid #eee;
    }

    // Portrait right
    &:not(&--left) {
        flex-direction: row-reverse;

        > .chat-item__data {
            margin-right: 1.25rem;
        }
    }

    // Portrait left
    &--left {
        > .chat-item__data {
            margin-left: 1.25rem;
        }
    }

    &__photo {
        background-color: #ff8b3e;
        border-radius: 50%;
        overflow: hidden;

        &,
        > img {
            height: 5rem;
            min-height: 5rem;
            min-width: 5rem;
            width: 5rem;
        }
    }

    &__data {
        padding-top: 0.313rem;
        width: 100%;
    }

    &__message {
        color: #777783;
        font-size: 0.875rem;
        margin-bottom: 0;
    }

    &__time {
        color: #9fadbb;
        font-size: 0.750rem;
        margin-left: 0.313rem;
    }

    &__name {
        color: #5f6468;
        font-size: 0.938rem;
        font-weight: 700;
    }

    &__time,
    &__name {
        line-height: 1;
    }

    &__upper {
        align-items: baseline;
        display: flex;
        margin-bottom: 0.313rem;
    }
}
