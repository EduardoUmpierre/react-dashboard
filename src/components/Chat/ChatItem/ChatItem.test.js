import React from 'react'
import { shallow } from 'enzyme'
import ChatItem from './ChatItem'

describe('<ChatItem />', () => {
    it('renders correctly', () => {
        const component = shallow(
            <ChatItem
                displayPortraitLeft
                message='Test'
                portrait=''
                time='1 min ago'
                userName='Eu'
            />
        )

        expect(component).toMatchSnapshot()
    })

    it('portrait is on the left', () => {
        const component = shallow(
            <ChatItem
                displayPortraitLeft
                message='Test'
                portrait=''
                time='1 min ago'
                userName='Eu'
            />
        )

        expect(component.find('.chat-item').hasClass('chat-item--left')).toBe(
            true
        )
    })

    it('portrait is on the right', () => {
        const component = shallow(
            <ChatItem
                displayPortraitLeft={false}
                message='Test'
                portrait=''
                time='1 min ago'
                userName='Eu'
            />
        )

        expect(component.find('.chat-item').hasClass('chat-item--left')).toBe(
            false
        )
    })
})
