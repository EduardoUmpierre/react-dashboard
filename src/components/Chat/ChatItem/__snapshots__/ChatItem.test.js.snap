// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatItem /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <chatItem
    displayPortraitLeft={true}
    message="Test"
    portrait=""
    time="1 min ago"
    userName="Eu"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="chat-item__photo"
        >
          
        </div>,
        <div
          className="chat-item__data"
        >
          <div
            className="chat-item__upper"
          >
            <div
              className="chat-item__name"
            >
              Eu
            </div>
            <div
              className="chat-item__time"
            >
              1 min ago
            </div>
          </div>
          <p
            className="chat-item__message"
          >
            Test
          </p>
        </div>,
      ],
      "className": "chat-item chat-item--left",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "chat-item__photo",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="chat-item__upper"
            >
              <div
                className="chat-item__name"
              >
                Eu
              </div>
              <div
                className="chat-item__time"
              >
                1 min ago
              </div>
            </div>,
            <p
              className="chat-item__message"
            >
              Test
            </p>,
          ],
          "className": "chat-item__data",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="chat-item__name"
                >
                  Eu
                </div>,
                <div
                  className="chat-item__time"
                >
                  1 min ago
                </div>,
              ],
              "className": "chat-item__upper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Eu",
                  "className": "chat-item__name",
                },
                "ref": null,
                "rendered": "Eu",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1 min ago",
                  "className": "chat-item__time",
                },
                "ref": null,
                "rendered": "1 min ago",
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test",
              "className": "chat-item__message",
            },
            "ref": null,
            "rendered": "Test",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="chat-item__photo"
          >
            
          </div>,
          <div
            className="chat-item__data"
          >
            <div
              className="chat-item__upper"
            >
              <div
                className="chat-item__name"
              >
                Eu
              </div>
              <div
                className="chat-item__time"
              >
                1 min ago
              </div>
            </div>
            <p
              className="chat-item__message"
            >
              Test
            </p>
          </div>,
        ],
        "className": "chat-item chat-item--left",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "chat-item__photo",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="chat-item__upper"
              >
                <div
                  className="chat-item__name"
                >
                  Eu
                </div>
                <div
                  className="chat-item__time"
                >
                  1 min ago
                </div>
              </div>,
              <p
                className="chat-item__message"
              >
                Test
              </p>,
            ],
            "className": "chat-item__data",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="chat-item__name"
                  >
                    Eu
                  </div>,
                  <div
                    className="chat-item__time"
                  >
                    1 min ago
                  </div>,
                ],
                "className": "chat-item__upper",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Eu",
                    "className": "chat-item__name",
                  },
                  "ref": null,
                  "rendered": "Eu",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1 min ago",
                    "className": "chat-item__time",
                  },
                  "ref": null,
                  "rendered": "1 min ago",
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test",
                "className": "chat-item__message",
              },
              "ref": null,
              "rendered": "Test",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
